MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/labs/RecyclerViewSymbolLayerActivity.java

```diff
@@ -0,0 +1,386 @@
+
+    // Initialize the map view
+    mapView = (MapView) findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(final MapboxMap mapboxMap) {
+
+        RecyclerViewSymbolLayerActivity.this.mapboxMap = mapboxMap;
+
+        Bitmap icon = BitmapFactory.decodeResource(
+          RecyclerViewSymbolLayerActivity.this.getResources(),
+          R.drawable.blue_marker_view);
+
+        mapboxMap.addImage("my-selected-marker-image", icon);
+
+        setUpLists();
+
+        mapView.addOnMapChangedListener(new MapView.OnMapChangedListener() {
+          @Override
+          public void onMapChanged(int change) {
+            if (change == WILL_START_RENDERING_MAP) {
+              addLayer(mapboxMap);
+            }
+          }
+        });
+
+        // Set up the recyclerView
+        locationAdapter = new LocationRecyclerViewAdapter(locationList, mapboxMap,
+          RecyclerViewSymbolLayerActivity.this);
+        recyclerView.setLayoutManager(new LinearLayoutManager(getApplicationContext(),
+          LinearLayoutManager.HORIZONTAL, true));
+        recyclerView.setItemAnimator(new DefaultItemAnimator());
+        recyclerView.setAdapter(locationAdapter);
+        SnapHelper snapHelper = new LinearSnapHelper();
+        snapHelper.attachToRecyclerView(recyclerView);
+        mapboxMap.setOnMapClickListener(RecyclerViewSymbolLayerActivity.this);
+      }
+    });
+  }
+
+  // Add the mapView lifecycle to the activity's lifecycle methods
+  @Override
+  public void onResume() {
+    super.onResume();
+    mapView.onResume();
+  }
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+    mapView.onStart();
+  }
+
+  @Override
+  protected void onStop() {
+    super.onStop();
+    mapView.onStop();
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    mapView.onPause();
+  }
+
+  @Override
+  public void onLowMemory() {
+    super.onLowMemory();
+    mapView.onLowMemory();
+  }
+
+  @Override
+  protected void onDestroy() {
+    super.onDestroy();
+    mapView.onDestroy();
+  }
+
+  @Override
+  protected void onSaveInstanceState(Bundle outState) {
+    super.onSaveInstanceState(outState);
+    mapView.onSaveInstanceState(outState);
+  }
+
+  private ArrayList<SingleRecyclerViewLocation> setUpLists() {
+
+    // Set up markers on the map and the location list to feed to the recyclerview
+
+    locationList = new ArrayList<>();
+
+    LatLng[] coordinates = new LatLng[] {
+      new LatLng(-34.6054099, -58.363654800000006),
+      new LatLng(-34.6041508, -58.38555650000001),
+      new LatLng(-34.6114412, -58.37808899999999),
+      new LatLng(-34.6097604, -58.382064000000014),
+      new LatLng(-34.596636, -58.373077999999964),
+      new LatLng(-34.590548, -58.38256609999996),
+      new LatLng(-34.5982127, -58.38110440000003)
+    };
+
+    for (int x = 0; x < 7; x++) {
```

Comment:
How about `coordinates.length` instead of `7`?

Reply:
Fixed